### PR TITLE
List as content item - sdk

### PR DESCRIPTION
**Pull request by Noy-Maimon**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/39603

## Description
Support Lists item in the following commands: **format**, **validate**, **download**, **upload**, **create-id-set**, **find-dependecies** and **create-content-artifacts**.

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -134,7 +137,8 @@ class FileType(Enum):
     FileType.GENERIC_DEFINITION: 'Objects',
     FileType.GENERIC_MODULE: 'Modules',
     FileType.GENERIC_TYPE: 'Object Types',
-    FileType.GENERIC_FIELD: 'Object Fields'
+    FileType.GENERIC_FIELD: 'Object Fields',
+    FileType.LISTS: 'Lists'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py:141]: [PEP8](https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas) here and everywhere else
```suggestion
    FileType.LISTS: 'Lists',
```

File: demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import demisto_client
+from wcmatch.pathlib import Path
+
+from demisto_sdk.commands.common.constants import LISTS, FileType
+from demisto_sdk.commands.common.content.objects.pack_objects.abstract_pack_objects.json_content_object import \
+    JSONContentObject
+
+
+class Lists(JSONContentObject):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py:11]: why is the name in plural? this object reffer to a single item

File: demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        """
+        Upload the lists item to demisto_client
+        Args:
+            client: The demisto_client object of the desired XSOAR machine to upload to.
+
+        Returns:
+            The result of the upload command from demisto_client
+        """
+        # return client.import_lists(file=self.path)
+        pass
```
Comment (yaakovi) [inline demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py:25]: you may implement it using the `generic_request_func` from the demisto-py client

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -299,9 +299,14 @@
...
     "invalid_generic_field_id": {'code': "GF101", 'ui_applicable': False, 'related_field': 'id'},
     "non_default_additional_info": {'code': "IN142", 'ui_applicable': True, 'related_field': 'additionalinfo'},
-    "missing_default_additional_info": {'code': "IN143", 'ui_applicable': True, 'related_field': 'additionalinfo'}
+    "missing_default_additional_info": {'code': "IN143", 'ui_applicable': True, 'related_field': 'additionalinfo'},
+    "invalid_from_server_version_in_lists": {'code': "L100", 'ui_applicable': False,
+                                             'related_field': 'fromServerVersion'},
+
 }
 
+# TODO: add errors
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:308]: ?

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+FROM_VERSION_LISTS = '6.5.0'
+
+
+class ListsValidator(ContentEntityValidator):
+    def __init__(self, structure_validator, ignored_errors=False, print_as_warnings=False,
+                 json_file_path=None, **kwargs):
+        super().__init__(structure_validator, ignored_errors, print_as_warnings,
+                         json_file_path=json_file_path, **kwargs)
+        self.from_version = self.current_file.get('fromServerVersion')
+        self.to_version = self.current_file.get('toServerVersion')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/lists.py:17]: for consistency with the rest of the content items
```suggestion
        self.from_version = self.current_file.get('fromVersion')
        self.to_version = self.current_file.get('toVersion')
```

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        """
+        return self._is_valid_version()
+
+    def is_valid_from_server_version(self) -> bool:
+        """Checks if from version field is valid.
+
+        Returns:
+            bool. True if from version field is valid, else False.
+        """
+        if self.from_version:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/lists.py:47]: if `from_version` is not defined, it should be considered as not valid.

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+            if LooseVersion(self.from_version) < LooseVersion(FROM_VERSION_LISTS):
+                error_message, error_code = Errors.invalid_from_server_version_in_lists('fromServerVersion')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/lists.py:49]:
```suggestion
                error_message, error_code = Errors.invalid_from_server_version_in_lists('fromVersion')
```

File: demisto_sdk/commands/common/constants.py [hunk 8]
```diff
@@ -532,7 +543,7 @@ class FileType(Enum):
     'incident_type',
     'layoutscontainer',
     'mapper',
-    'pre_process_rule',
+    'pre_process_rule'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py:546]: `list` should also be added here

File: demisto_sdk/commands/common/schemas/list.yml [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+  allReadWrite:
+    type: bool
+    required: True
+  data:
+    type: str
+  dbotCreatedBy:
+      type: str
+  description:
+    type: str
+  fromServerVersion:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/schemas/list.yml:15]:
```suggestion
  fromVersion:
```

File: demisto_sdk/commands/common/schemas/list.yml [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+    sequence:
+      - includes: roles_schema
+  system:
+    type: bool
+    required: True
+  tags:
+    type: seq
+    sequence:
+      - type: str
+  toServerVersion:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/schemas/list.yml:57]:
```suggestion
  toVersion:
```

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/lists.py:50]: add a suggested fix to use the format command.
see uses of `Errors.suggest_fix`.

File: demisto_sdk/commands/format/update_lists.py [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            return ERROR_RETURN_CODE
+
+    def set_from_server_version_to_default(self, location=None):
+        """Replaces the fromServerVersion of the YML to default."""
+        if self.verbose:
+            click.echo(f'Trying to set JSON fromServerVersion to default: {FROM_VERSION_LISTS}')
+        if location and not location.get('fromServerVersion'):
+            location['fromServerVersion'] = FROM_VERSION_LISTS
+        else:
+            self.data.setdefault('fromServerVersion', FROM_VERSION_LISTS)
```
Comment (yaakovi) [inline demisto_sdk/commands/format/update_lists.py:58]:
```suggestion
    def set_from_server_version_to_default(self, location=None):
        """Replaces the fromVersion of the YML to default."""
        if self.verbose:
            click.echo(f'Trying to set JSON fromVersion to default: {FROM_VERSION_LISTS}')
        if location and not location.get('fromVersion'):
            location['fromVersion'] = FROM_VERSION_LISTS
        else:
            self.data.setdefault('fromVersion', FROM_VERSION_LISTS)
```

File: demisto_sdk/commands/common/errors.py [hunk 13]
```diff
@@ -299,9 +299,14 @@
     "invalid_generic_field_group_value": {'code': "GF100", 'ui_applicable': False, 'related_field': 'group'},
     "invalid_generic_field_id": {'code': "GF101", 'ui_applicable': False, 'related_field': 'id'},
     "non_default_additional_info": {'code': "IN142", 'ui_applicable': True, 'related_field': 'additionalinfo'},
-    "missing_default_additional_info": {'code': "IN143", 'ui_applicable': True, 'related_field': 'additionalinfo'}
+    "missing_default_additional_info": {'code': "IN143", 'ui_applicable': True, 'related_field': 'additionalinfo'},
+    "invalid_from_server_version_in_lists": {'code': "L100", 'ui_applicable': False,
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:303]: error codes should have 2 letter prefix
```suggestion
    "invalid_from_server_version_in_lists": {'code': "LI100", 'ui_applicable': False,
```

File: demisto_sdk/tests/test_files/artifacts/content/content_packs/Sample01/PreProcessRules/preprocessrule-Drop.json [hunk 14]
```diff
@@ -0,0 +1,69 @@
+{
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/artifacts/content/content_packs/Sample01/PreProcessRules/preprocessrule-Drop.json:1]: why did you add this file?

File: demisto_sdk/tests/integration_tests/find_dependencies_integration_test.py [hunk 15]
```diff
@@ -56,6 +57,7 @@ def test_integration_find_dependencies_sanity(self, mocker, repo):
         mocker.patch.dict(os.environ, {'DEMISTO_SDK_ID_SET_REFRESH_INTERVAL': '-1'})
         pack = repo.create_pack('FindDependencyPack')
         integration = pack.create_integration('integration')
+        integration.create_default_integration()
```
Comment (yaakovi) [inline demisto_sdk/tests/integration_tests/find_dependencies_integration_test.py:60]: why did you modify this test?

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md [hunk 16]
```diff
@@ -59,4 +59,94 @@ If you are interested in contributing, visit our contribution process [here](htt
 
 Thank you for contributing to Cortex XSOAR.
 
+If you are interested in contributing, visit our contribution process [here](https://xsoar.pan.dev/docs/contributing/contributing).#### This pack was co-authored by:
```
Comment (yaakovi) [inline demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:62]: why did you modify this file?

**Review by yaakovi — CHANGES_REQUESTED** [review]
nice work

File: demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py (hunk 2 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py:11]: The class `List()` was shadowing the builtin python class, and caused errors while running, so I thought `Lists()` is better than `ListItem()` or anything similar. What do you think?

File: demisto_sdk/commands/common/hook_validations/lists.py (hunk 5 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/hook_validations/lists.py:17]: Currently the lists example I was working with had the fromServerVersion field, see attached example
image

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md (hunk 16 above)
Comment (Noy-Maimon) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:62]: Not sure how this was modified, reverting

File: demisto_sdk/commands/common/hook_validations/lists.py (hunk 11 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/hook_validations/lists.py:50]: Done

File: demisto_sdk/commands/common/constants.py [hunk 17]
```diff
@@ -159,7 +163,9 @@ class FileType(Enum):
     FileType.GENERIC_DEFINITION.value: GENERIC_DEFINITIONS_DIR,
     FileType.GENERIC_MODULE.value: GENERIC_MODULES_DIR,
     FileType.GENERIC_FIELD.value: GENERIC_FIELDS_DIR,
-    FileType.GENERIC_TYPE.value: GENERIC_TYPES_DIR
+    FileType.GENERIC_TYPE.value: GENERIC_TYPES_DIR,
+    FileType.LISTS.value: LISTS_DIR
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py:167]:
```suggestion
    FileType.LISTS.value: LISTS_DIR,
```

Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py]:
```suggestion
    FileType.LISTS.value: LISTS_DIR,
```

File: demisto_sdk/commands/common/constants.py [hunk 18]
```diff
@@ -186,6 +192,7 @@ class FileType(Enum):
     GENERIC_MODULES_DIR,
     GENERIC_DEFINITIONS_DIR,
     PRE_PROCESS_RULES_DIR,
+    LISTS_DIR
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py]:
```suggestion
    LISTS_DIR,
```

File: demisto_sdk/commands/common/constants.py [hunk 19]
```diff
@@ -202,6 +209,7 @@ class FileType(Enum):
     LAYOUTS_DIR,
     DASHBOARDS_DIR,
     PRE_PROCESS_RULES_DIR,
+    LISTS_DIR
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py:212]:
```suggestion
    LISTS_DIR,
```

File: demisto_sdk/commands/common/constants.py [hunk 20]
```diff
@@ -787,6 +799,7 @@ class FileType(Enum):
     INDICATOR_TYPES_DIR,
     CONNECTIONS_DIR,
     INDICATOR_FIELDS_DIR,
+    LISTS_DIR
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py:802]:
```suggestion
    LISTS_DIR,
```

File: demisto_sdk/commands/common/constants.py [hunk 21]
```diff
@@ -1180,6 +1193,7 @@ class PB_Status:
     'Classifiers',
     'Layouts',
     'PreProcessRules',
+    'Lists'
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py:1196]:
```suggestion
    'Lists',
```

File: demisto_sdk/commands/common/constants.py [hunk 22]
```diff
@@ -1279,7 +1294,8 @@ class ContentItems(Enum):
     LAYOUTS_DIR,
     PRE_PROCESS_RULES_DIR,
     CLASSIFIERS_DIR,
-    WIDGETS_DIR
+    WIDGETS_DIR,
+    LISTS_DIR
```
Comment (Noy-Maimon) [inline demisto_sdk/commands/common/constants.py:1298]:
```suggestion
    LISTS_DIR,
```

File: demisto_sdk/tests/integration_tests/find_dependencies_integration_test.py (hunk 15 above)
Comment (Noy-Maimon) [reply demisto_sdk/tests/integration_tests/find_dependencies_integration_test.py:60]: The current test is not passing on master branch (not sure how this is possible), but adding this line, which populates the created integration in the previous line, assures the test passes. This was added based on this [PR](https://github.com/demisto/demisto-sdk/pull/1606/files) (which is pending merge).

File: demisto_sdk/commands/common/constants.py [hunk 23]
```diff
@@ -43,6 +44,7 @@
 LAYOUT = 'layout'
 LAYOUTS_CONTAINER = 'layoutscontainer'
 PRE_PROCESS_RULES = 'pre-process-rules'
+LISTS = 'list'  # singular, as it is the prefix of the file
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/constants.py:47]: Wouldn't List as name be better?
List = 'list'
or did I miss something here (as it represents a single List object, not Lists object)

File: demisto_sdk/commands/common/constants.py [hunk 24]
```diff
@@ -110,6 +112,7 @@ class FileType(Enum):
     GENERIC_TYPE = 'generictype'
     GENERIC_DEFINITION = 'genericdefinition'
     PRE_PROCESS_RULES = 'pre-process-rule'
+    LISTS = 'list'
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/constants.py:115]: same here

File: demisto_sdk/commands/common/constants.py [hunk 25]
```diff
@@ -134,7 +137,8 @@ class FileType(Enum):
     FileType.GENERIC_DEFINITION: 'Objects',
     FileType.GENERIC_MODULE: 'Modules',
     FileType.GENERIC_TYPE: 'Object Types',
-    FileType.GENERIC_FIELD: 'Object Fields'
+    FileType.GENERIC_FIELD: 'Object Fields',
+    FileType.LISTS: 'Lists',
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/constants.py:141]: same here

File: demisto_sdk/commands/common/constants.py [hunk 26]
```diff
@@ -407,6 +415,9 @@ class FileType(Enum):
 PACKS_PRE_PROCESS_RULES_DIR_REGEX = fr'{PACK_DIR_REGEX}/{PRE_PROCESS_RULES_DIR}'
 PACKS_PRE_PROCESS_RULES_JSON_REGEX = fr'{PACKS_PRE_PROCESS_RULES_DIR_REGEX}/(?:preprocessrule-)?([^/]+)\.json'
 
+PACKS_LISTS_DIR_REGEX = fr'{PACK_DIR_REGEX}/{LISTS_DIR}'
+PACKS_LISTS_JSON_REGEX = fr'{PACKS_LISTS_DIR_REGEX}/(?:list-)?([^/]+)\.json'
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/constants.py:419]: Why do we need the list- as a possible for regex? just catch everything inside the dir

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 27]
```diff
@@ -0,0 +1,54 @@
...
+        """
+        # Lists files have fromServerVersion instead of fromVersion
+        validations: List = [
+            self.is_valid_version(),
+            self.is_valid_from_server_version(),
+        ]
+
+        return all(validations)
+
+    def is_valid_version(self) -> bool:
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/hook_validations/lists.py:33]: just call to _is_valid_version from is_valid_list, this function that calls a function isn't really needed

File: demisto_sdk/commands/common/tests/lists_test.py [hunk 28]
```diff
@@ -0,0 +1,47 @@
...
+        structure.prev_ver = 'master'
+        structure.branch_name = ''
+        structure.quite_bc = False
+        return structure
+
+
+class TestListValidator:
+
+    @pytest.mark.parametrize('list_path, is_valid', [(LIST_GOOD, True)])
+    def test_is_valid_list(self, list_path, is_valid):
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/tests/lists_test.py:36]: add tests for all cases, (fromVersion, fromServerVersion, toVersion, toServerVersion), and invalid cases as well

File: demisto_sdk/commands/common/tools.py [hunk 29]
```diff
@@ -1195,6 +1195,9 @@ def find_type(path: str = '', _dict=None, file_type: Optional[str] = None, ignor
                 'newEventFilters' in _dict and 'readyNewEventFilters' in _dict:
             return FileType.PRE_PROCESS_RULES
 
+        if 'allRead' in _dict and 'truncated' in _dict:
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/tools.py:1198]: Please also add to `find_type_by_path` if the file ends with json and is in lists dir as well

File: demisto_sdk/commands/download/tests/downloader_test.py [hunk 30]
```diff
@@ -42,6 +42,8 @@ def __init__(self, tmp_path):
         self.SCRIPT_INSTANCE_PATH = None
         self.PLAYBOOK_INSTANCE_PATH = None
         self.LAYOUT_INSTANCE_PATH = None
+        self.PRE_PROCESS_RULES_INSTANCE_PATH = None
```
Comment (tomneeman151293) [inline demisto_sdk/commands/download/tests/downloader_test.py:45]: How is this related?

File: demisto_sdk/commands/download/tests/downloader_test.py [hunk 31]
```diff
@@ -122,7 +126,11 @@ def setup(self):
         ]}
         self.PRE_PROCESS_RULES_PACK_OBJECT = {'DummyPreProcessRule': [
             {'name': 'DummyPreProcessRule', 'id': 'DummyPreProcessRule',
-             'path': self.PRE_PROCESS_RULES_INSTANCE_PATH, 'file_ending': 'yml'}
+             'path': self.PRE_PROCESS_RULES_INSTANCE_PATH, 'file_ending': 'yml'}  # TODO: shouldnt be json?
```
Comment (tomneeman151293) [inline demisto_sdk/commands/download/tests/downloader_test.py:129]: solve TODO

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md (hunk 16 above)
Comment (tomneeman151293) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:62]: Seems this was not reverted

File: demisto_sdk/commands/common/update_id_set.py [hunk 32]
```diff
@@ -144,6 +145,30 @@ def build_tasks_graph(playbook_data):
     return graph
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/update_id_set.py:147]: Need to consider cases of integrations or scripts callng getList and setList

File: demisto_sdk/commands/common/update_id_set.py [hunk 33]
```diff
@@ -483,6 +509,8 @@ def get_playbook_data(file_path: str) -> dict:
         playbook_data['filters'] = filters
     if transformers:
         playbook_data['transformers'] = transformers
+    if implementing_lists:
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/update_id_set.py:512]: as said above, also for integrations/scripts

File: demisto_sdk/commands/common/update_id_set.py [hunk 34]
```diff
@@ -1378,6 +1407,17 @@ def get_generic_module_data(path):
     return {id_: data}
 
 
+def get_list_data(path: str):
+    json_data = get_json(path)
+    data = create_common_entity_data(path=path,
+                                     name=json_data.get('name'),
+                                     to_version=json_data.get('toServerVersion'),
+                                     from_version=json_data.get('fromServerVersion'),
```
Comment (tomneeman151293) [inline demisto_sdk/commands/common/update_id_set.py:1415]: could there be a situation of toVersion/fromVersion in json_data instead?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 35]
```diff
@@ -514,7 +514,14 @@ def _collect_playbooks_dependencies(pack_playbooks: list, id_set: dict, verbose:
                 playbook_data.get('implementing_playbooks', []),
                 skippable_tasks,
                 id_set['playbooks'],
-                exclude_ignored_dependencies
+                exclude_ignored_dependencies,
+            ))
+
+            playbook_dependencies.update(PackDependencies._differentiate_playbook_implementing_objects(
```
Comment (tomneeman151293) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:520]: Same as said in ID set, need to consider dependencies for integrations/scripts calling getList/setList (if They're found in the ID set of course).

Probably only getList is important, setList shouldn't have dependency I guess

File: demisto_sdk/commands/format/update_lists.py [hunk 36]
```diff
@@ -0,0 +1,58 @@
...
+from typing import Tuple
+
+import click
+
+from demisto_sdk.commands.format.format_constants import (ERROR_RETURN_CODE,
+                                                          SKIP_RETURN_CODE,
+                                                          SUCCESS_RETURN_CODE)
+from demisto_sdk.commands.format.update_generic_json import BaseUpdateJSON
+
+FROM_VERSION_LISTS = '6.5.0'
```
Comment (tomneeman151293) [inline demisto_sdk/commands/format/update_lists.py:11]:
```suggestion
MIN_FROM_VERSION_LISTS = '6.5.0'
```

File: demisto_sdk/commands/download/tests/downloader_test.py (hunk 30 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/download/tests/downloader_test.py:45]: It was missing for `pre-processr-rule`, so I added it

File: demisto_sdk/commands/common/tools.py (hunk 29 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tools.py:1198]: Done

File: demisto_sdk/commands/common/tests/lists_test.py (hunk 28 above)
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/tests/lists_test.py:36]: Done

File: demisto_sdk/commands/common/hook_validations/lists.py [hunk 37]
```diff
@@ -31,12 +32,14 @@ def is_valid_list(self) -> bool:
         return all(validations)
 
     def is_valid_version(self) -> bool:
```
[the remark this answers is not in the document]
Comment (Noy-Maimon) [reply demisto_sdk/commands/common/hook_validations/lists.py:34]: Done

File: demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py [hunk 38]
```diff
@@ -21,8 +22,8 @@ def upload(self, client: demisto_client):
         Returns:
             The result of the upload command from demisto_client
         """
-        # return client.import_lists(file=self.path)
-        pass
+        return demisto_client.generic_request_func(self=client, path='lists/save',
+                                                   body=json.dumps(self._as_dict), method='POST')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/content/objects/pack_objects/lists/lists.py:26]:
```suggestion
        return client.generic_request_func(
            method='POST',
            path='lists/save',
            body=json.dumps(self._as_dict),
        )
```
